### ISSUE TITLE
Capture agent options should be pre-selected when scheduling a recording

### ISSUE THREAD

**Issue by snoesberger** [opencast/opencast-admin-interface#941]
When I schedule a new capture event recording, the Extron SMP options should be pre-selected in the scheduling dialogue:
![grafik](https://github.com/user-attachments/assets/74f8fc87-4e0b-489a-8437-7f16470cbde5)

**Comment by mtneug** [comment]
I would bet, some people want the exact opposite behaviour. pyCA doesn't support this (IIRC) so I may be misinformed, but I always felt like this part of the CA API is underspecified. E.g. what does it mean, if no input is selected?

**Comment by snoesberger** [comment]
> I would bet, some people want the exact opposite behaviour.

I don't think so. If different inputs are available for the CA (and can be configured), at least one of the inputs must be selected by the user. So at least one click is required. At our university, we always record both channels. Later, in the editor, the user can choose to publish both videos or just one. For us, leaving these checkboxes unchecked by default means that the user has to manually select both channels each time they plan a new recording (with the risk of forgetting to do so). We therefore prefer to have the input channels pre-selected.

> E.g. what does it mean, if no input is selected?

In the case of Extron SMP CAs, the video from the unselected channel is muted, only a black picture is recorded (and not available in the live stream).

**Comment by mtneug** [comment]
In that case, shouldn't it be a required option? Should Opencast check that at least one checkbox is selected? IMO, it would be better to force the user to make a selection instead of making this implicit.

**Comment by gregorydlogan** [comment]
It used to be a requirement that at least one input was selected. IMO this is a bug in a few places: the backend should validate that at least one is selected, the front end should do the same, and the ui should support having them selected by default (tho I agree with @mtneug here)

**Comment by Arnei** [comment]
To summarize:
- All inputs should be preselected (set to true).
- At least one input must be selected before proceeding.
- If a CA has no inputs, ignore the above.

Is that about right?

**Comment by marcusm7** [comment]
@Arnei your summary sounds about right for our case (Uni Konstanz).

1. Maybe introduce a flag in the configuration which allows to define the input-preselected behaviour?  

2. One step further: a non-admin-user of the admin-ui  might not know what channel a/b means - the established names seem to be: 
- slides or presentation and 
- presenter or camera  
  
  So an optional configurable key value map might be nice, what is displayed for the user instead of the raw channel names of the CA (i.e CAPTURE_AGENT.DEVICE.CHANNEL A -> Slides). But that is just an idea. We don't have end users in the admin ui and our team knows what channel a and b means.

**Comment by mtneug** [comment]
I think A/B naming comes from the CA? In any case, Opencast does support more than two videos which should be supported by such a mapping.

**Comment by gregorydlogan** [comment]
The CA is responsible to populating those fields yes.  You can, in theory, overwrite what the CA sends by POSTing the right bits to the CA admin endpoints, but if the CA periodically resends its capabilities (which it's supposed to) then your changes would get overwritten at that point :)

**Comment by reiferschris** [comment]
A few notes:

- +1 for having the inputs preselected by default. 
- A configuration option is nice but more important: having no inputs selected should fail in some form validation
- @snoesberger screenshot shows that the new admin-ui doesn't use the `capture.device.names` to display the input names. In case of extron smps that's `Channel A,Channel B` which were displayed before. 
- personally I'm not a fan of such a mapping proposed by @marcusm7 It tries to solve a problem, that is introduced by another party (in this case extron). If people are collectively bothered by this -> write extron, they always state that this is the only way they will listen
- sidenote: for anyone interested, pyca can handle inputs with this PR https://github.com/opencast/pyCA/pull/413. You can even name them whatever you want, and they show up like this in the admin ui (we have presenter, presentation and blackboard). Also the epiphan devices allow choosing input names you desire.

**Comment by snoesberger** [comment]
I completely agree with @reiferschris and would like to summarise my point of view:
- I'm still in favour of the inputs being preselected by default.
- At least one input must be selected before proceeding.
- If a CA has no inputs, ignore the above.
- Use capture.device.names to display the input names.
- The default selected/not-selected configuration option is nice, but not a must-have.
- The input naming problem should be solved by the CA manufacturer.
